kahuna/public/js/leases/leases.js

```diff
@@ -31,60 +47,81 @@ leases.controller(
                 $window.alert('Please select an access type (Allow or Deny)');
             } else {
                 ctrl.adding = true;
-                ctrl.newLease.mediaId = ctrl.image.data.id;
                 ctrl.newLease.createdAt = new Date();
                 ctrl.newLease.access = ctrl.access;
 
-                leaseService.add(ctrl.image, ctrl.newLease)
-                    .then(() => {
-                        ctrl.updateLeases(ctrl.image);
-                    })
+                leaseService.batchAdd(ctrl.newLease, ctrl.leases, ctrl.images)
                     .catch(() =>
-                        alertFailed('Something went wrong when saving, please try again!')
+                        alertFailed('Something went wrong when saving, please try again.')
                     )
                     .finally(() => {
-                        ctrl.editing = false;
-                        ctrl.adding = false;
                         ctrl.resetLeaseForm();
                 });
             }
         };
 
+        function addLease(newLease) {
```

Comment:
When called from a batch event we can't distinguish between the image that originated the update and the ones to which it _should_ apply. Consequently the lease is double applied to the originating image.